dpnp/dpnp_iface_arraycreation.py

```diff
@@ -697,14 +704,15 @@ def full(shape, fill_value, dtype=None, order='C'):
     [10, 10, 10, 10]
 
     """
-    if not use_origin_backend():
-        if order not in ('C', 'c', None):
-            pass
-        else:
-            if dtype is None:
-                dtype = numpy.array(fill_value).dtype.type  # TODO simplify
-
-            return dpnp_full(shape, fill_value, dtype).get_pyobj()
+    if like is None:
+        if dtype is None:
```

Comment:
This case work with  fill_value=xp.array(1, dtype=dtype) and cannot be removed

Reply:
I believe, dpctl should support `fill_value` as `usm_ndarray` in case `dtype` is `None` and such logic needs to be moved into dpctl code, since NumPy supports the same case.
While in dpnp we need to add a check to `dpnp_container.py` to explicitly pass `fill_value` as `usm_ndarray`, like:
```
    """Converts `fill_value` to `usm_ndarray`"""
    value = fill_value.get_array() if isinstance(fill_value, dpnp_array) else fill_value

    """Creates `dpnp_array` where every element is equal to fill_value."""
    array_obj = dpt.full(shape,
                         value,
                         dtype=dtype,
                         order=order,
                         usm_type=usm_type,
                         sycl_queue=sycl_queue_normalized)
```